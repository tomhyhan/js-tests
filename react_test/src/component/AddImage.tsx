import React from 'react'

export default function AddImage({onAddImage}: {
    onAddImage: (imageLayout:string) => void
}) {
    const handleImageAdd = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        const formData = new FormData(e.currentTarget);
        const imageLayout = formData.get('imageLayout')
        onAddImage(imageLayout!.toString())
    }

  return (
    <form className="flex p-5" onSubmit={handleImageAdd}>
        <select name="imageLayout" defaultValue="normal" className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded block mr-5">
            <option value="normal">normal</option>
            <option value="blur">blur</option>
            <option value="grayscale">grayscale</option>
        </select>
        <button type="submit" className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Add Image</button>
    </form>
  )
}
