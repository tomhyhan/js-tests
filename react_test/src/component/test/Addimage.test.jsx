import { render, screen } from "@testing-library/react"
import {afterEach, describe, expect, it, vi} from "vitest"
import AddImage from './../AddImage';
import userEvent from "@testing-library/user-event";

describe("Addimage", () => {
    const onAddImage = vi.fn();
    
    afterEach(() => {
        onAddImage.mockReset()
    })

    it("renders Addimage correctly", async () => {
        const { asFragment } = render(<AddImage onAddImage={onAddImage}/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it("calls onAddImage when form is submitted", async () => {
        render(<AddImage onAddImage={onAddImage}/>)

        const button = screen.getByRole("button")
        await userEvent.click(button)
        expect(onAddImage).toHaveBeenCalledOnce()
    })
    
})