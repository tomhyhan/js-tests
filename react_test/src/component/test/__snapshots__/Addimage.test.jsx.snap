// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Addimage > renders Addimage correctly 1`] = `
<DocumentFragment>
  <form
    class="flex p-5"
  >
    <select
      class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded block mr-5"
      name="imageLayout"
    >
      <option
        selected=""
        value="normal"
      >
        normal
      </option>
      <option
        value="blur"
      >
        blur
      </option>
      <option
        value="grayscale"
      >
        grayscale
      </option>
    </select>
    <button
      class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
      type="submit"
    >
      Add Image
    </button>
  </form>
</DocumentFragment>
`;
